### PR TITLE
added controllers & policies for pundit

### LINKED ISSUE #3: Authorization with Pundit
Add pundit authorization to increase security

### PR DESCRIPTION
**Pull request by MauritzKurt**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,12 +1,27 @@
 class ApplicationController < ActionController::Base
+  include Pundit
+
   before_action :authenticate_user!
 
   before_action :configure_permitted_parameters, if: :devise_controller?
 
+  after_action :verify_authorized, unless: :devise_controller?
```
Comment (ajmompr) [inline app/controllers/application_controller.rb:7]: good job accounting for devise

File: app/controllers/likes_controller.rb
Comment (ajmompr) [inline app/controllers/likes_controller.rb:1]: no need to add pundit to likes for this assignment

**Review by ajmompr — CHANGES_REQUESTED** [review]
Hi Kurt! Let's get together to review Pundit this afternoon! There are quite a few changes needed

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -59,21 +47,25 @@ def destroy
...
+    def comment_params
+      params.require(:comment).permit(:author_id, :photo_id, :body)
     end
-  end
 
-  # Only allow a list of trusted parameters through.
-  def comment_params
-    params.require(:comment).permit(:author_id, :photo_id, :body)
-  end
+    def is_an_authorized_user
```
Comment (heratyian) [inline app/controllers/comments_controller.rb]: this should be handled by CommentPolicy

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -57,13 +50,24 @@ def destroy
...
+
+  def set_like
+    @like = Like.find(params[:id])
+  end
+
+  def authorize_resource
+    authorize @like || Like
+  end
+
+  def authorize_create_action
```
Comment (heratyian) [inline app/controllers/likes_controller.rb:62]: i don't think you need this method abstraction.

Comment (MauritzKurt) [reply app/controllers/likes_controller.rb:62]: removed

File: app/controllers/likes_controller.rb
Comment (MauritzKurt) [reply app/controllers/likes_controller.rb:1]: removed

**Review by ajmompr — APPROVED** [review]

**Pull request merged**